### PR TITLE
chore: better parser errors

### PR DESCRIPTION
**Pull request by azjezz**

### CODE REVIEW

**Comment by azjezz** [comment]
@ryangjchandler i wanted to move all tests using `assert_ast` to integration tests, but i think this PR is also big enough as is :) will do that on the next PR.

**Review by ryangjchandler — CHANGES_REQUESTED** [review]
This is a solid refactor, the only thing that doesn't feel right to me is the order of args to the `expect_token!` macro. My mind would prefer to read it in this order, I think:

`expect_token!(expected tokens go here, parser, message)`.. what do you think?

**Comment by azjezz** [comment]
sounds good! will do :)

**Comment by azjezz** [comment]
done.

File: src/parser/ident.rs [hunk 1]
```diff
@@ -1,47 +1,37 @@
...
     pub(crate) fn ident(&mut self) -> ParseResult<ByteString> {
-        Ok(expect!(
-            self,
-            TokenKind::Identifier(i),
-            i,
-            "expected identifier"
-        ))
+        Ok(expect_token!([
+            TokenKind::Identifier(identifier) => identifier,
+        ], self, "an identifier"))
```
Comment (azjezz) [inline src/parser/ident.rs:11]: i flipped the pattern and value on purpose, this now looks more like rust code :)

Comment (ryangjchandler) [reply src/parser/ident.rs:11]: yep, this is much nicer

**Review by ryangjchandler — APPROVED** [review]

**Pull request merged**